clients/src/test/java/org/apache/kafka/common/requests/RequestResponseTest.java

```diff
@@ -1269,24 +909,656 @@ public void testDeletableTopicResultErrorMessageIsNullByDefault() {
+                .setPartitions(asList(0, 1, 2))
+                .setTopic("topic1"));
+        DescribeLogDirsRequestData data = new DescribeLogDirsRequestData()
+                .setTopics(topics);
+        return new DescribeLogDirsRequest.Builder(data).build(version);
+    }
+
+    private DescribeLogDirsResponse createDescribeLogDirsResponse() {
+        DescribeLogDirsResponseData data = new DescribeLogDirsResponseData()
+                .setResults(singletonList(new DescribeLogDirsResponseData.DescribeLogDirsResult()
+                        .setErrorCode(Errors.NONE.code())
+                        .setLogDir("logdir")
+                        .setTopics(singletonList(new DescribeLogDirsResponseData.DescribeLogDirsTopic()
+                                .setName("topic1")
+                                .setPartitions(singletonList(new DescribeLogDirsResponseData.DescribeLogDirsPartition()
+                                        .setPartitionIndex(0)
+                                        .setIsFutureKey(false)
+                                        .setOffsetLag(123L)
+                                        .setPartitionSize(234L)))))))
+                .setThrottleTimeMs(123);
+        return new DescribeLogDirsResponse(data);
+    }
+
+    private DeleteRecordsRequest createDeleteRecordsRequest(short version) {
+        DeleteRecordsRequestData.DeleteRecordsTopic topic = new DeleteRecordsRequestData.DeleteRecordsTopic()
+                .setName("topic1")
+                .setPartitions(singletonList(new DeleteRecordsRequestData.DeleteRecordsPartition()
+                        .setPartitionIndex(1)
+                        .setOffset(123L)));
+        DeleteRecordsRequestData data = new DeleteRecordsRequestData()
+                .setTopics(singletonList(topic))
+                .setTimeoutMs(123);
+        return new DeleteRecordsRequest.Builder(data).build(version);
+    }
+
+    private DeleteRecordsResponse createDeleteRecordsResponse() {
+        DeleteRecordsResponseData.DeleteRecordsTopicResultCollection topics = new DeleteRecordsResponseData.DeleteRecordsTopicResultCollection();
+        DeleteRecordsResponseData.DeleteRecordsPartitionResultCollection partitions = new DeleteRecordsResponseData.DeleteRecordsPartitionResultCollection();
+        partitions.add(new DeleteRecordsResponseData.DeleteRecordsPartitionResult()
+                .setErrorCode(Errors.NONE.code())
+                .setLowWatermark(123L)
+                .setPartitionIndex(0));
+        topics.add(new DeleteRecordsResponseData.DeleteRecordsTopicResult()
+                .setName("topic1")
+                .setPartitions(partitions));
+        DeleteRecordsResponseData data = new DeleteRecordsResponseData()
+                .setThrottleTimeMs(123)
+                .setTopics(topics);
+        return new DeleteRecordsResponse(data);
+    }
+
+    private DescribeClusterRequest createDescribeClusterRequest(short version) {
```

Comment:
I see that many methods use a `short` for the version, but plenty still use `int`. Perhaps we should standardize?

Reply:
What's the benefit of using `short` over `int`? It means we have to do a lot more type casts, but doesn't make a difference in terms of test coverage.

Reply:
Actually, I suppose it's not as bad as I first thought. We're removing more casts than we're adding.

Reply:
Yeah I thought this was a bit better like this.